### ISSUE TITLE
On smaller widths, it's still possible to overlap dates on the x-axis

### ISSUE THREAD

**Issue by devmi** [mozilla/metrics-graphics#762]
On a smaller width (e.g. 400px), it's still possible to overlap dates on x-axis on **v2.11**.
<img width="381" alt="mg-issue-561" src="https://user-images.githubusercontent.com/19313261/27504380-4b6c99ee-583d-11e7-88db-95d5401d77be.png">

To reproduce it, please copy the [auto time formatting > Over A Large Span of Days](https://www.metricsgraphicsjs.org/examples.htm#auto-time-formatting) sample code to the [interactive demo](https://www.metricsgraphicsjs.org/interactive-demo.htm) demo, adjusting some values as shown below:

```javascript
function fake_days(length) {  //no changes
    var d = new Date();
    var v = 100000;

    var data = [];
    for (var i = 0; i < length; i++) {
        v += (Math.random() - 0.5) * 10000;
        if (v < 0) v = 0;
        data.push({date: MG.clone(d), value: v});
        d.setDate(d.getDate() + 1);
    }
    return data;
}
var many_days = fake_days(80);  //also reproducible w/other values, like 23, 81,82...

MG.data_graphic({
    title: "Over A Large Span of Days",
    data: many_days,
    target: ".result",  //'#time4',  //set the target to the sample class
    width: 400,   //600,    //make the width smaller
    height: 200,
    right: 40
});
```
 
---
issue created from [this comment](https://github.com/mozilla/metrics-graphics/issues/561#issuecomment-310803128)

**Comment by wlach** [comment]
I have noticed this behaviour as well, would be nice to fix.

**Comment by himanish-star** [comment]
@wlach, can I take over this issue?

**Comment by himanish-star** [comment]
@hamilton, I need some help with this issue. Where are these markers set from(file)? I tried logging inside all the related functions inside `src/js/common/x_axis.js` but didn't notice anything. I have followed the instructions mentioned in https://github.com/mozilla/metrics-graphics/issues/762#issue-240743931. I think the changes have to be made in `x_axis` but I may be wrong. Please guide me :smiley:

**Comment by wlach** [comment]
@himanish-star how are you testing? I recently made it a bit easier to get a test environment up and running: https://github.com/mozilla/metrics-graphics#contributing

And yes, you are more than welcome at trying your hand at fixing this issue. :) Or really any other that no one has claimed to be working on (or hasn't seen activity in a while).

**Comment by himanish-star** [comment]
>I recently made it a bit easier to get a test environment up and running:

Yes this is how I'm testing it. I want to know that where are the `labels` for the x-axis set. When I was inspecting the x-axis in the browser inspector then I found out that `<line>` and `<text>` elements were used in the x-axis. So I started searching for this and I found these elements inside the `x_axis.js` file. But as I said when I logged into most of the functions inside this file which I thought may be relative to setting the markers of the x_axis, then I found nothing. Thus this means that my search was not right. That is why I need some help regarding this

**Comment by himanish-star** [comment]
@wlach, what is the `id` of the IRC channel of `metrics-graphics`

**Comment by wlach** [comment]
@himanish-star I just created a channel on irc.mozilla.org: #metricsgraphics. Feel free to ask questions there.

To be honest I am still getting to grips with the code as well, but I think maybe a search for ticks might be productive at helping you find where to look for things to log/change. I get a lot of hits for that with `git grep` which seem promising.

**Comment by himanish-star** [comment]
Thanks. I think I have found a solution to this issue. And thanks for creating an IRC channel. :smiley:
